closure/goog/dom/xml.js

```diff
@@ -39,6 +39,33 @@ goog.dom.xml.MAX_ELEMENT_DEPTH = 256; // Same default as MSXML6.
 
 
 /**
+ * Check for ActiveXObject support by the browser.
+ * @return {boolean} true if browser has ActiveXObject support.
+ * @private
+ */
+goog.dom.xml.hasActiveXObjectSupport_ = function() {
+  try {
+    // Due to lot of changes in IE 9, 10 & 11 behaviour, trying to create the
+    // ActiveXOjbect is a lot more reliable than testing for the existance of
+    // window.ActiveXObject
+    new ActiveXObject('MSXML2.DOMDocument');
```

Comment:
I'm going to get complaints about this as folks like code to run exception clean so they can "pause on throw" in the debuggers.  Also, for compiled code, correct behavior here relies on creating ActiveXObject being detected as having side-effects which as far as I know isn't actually the case.  Also if it isn't eliminated, calling this in global scope means the code will never be eliminated.

Generally, I think using using goog.userAgent.IE will be a lot cleaner, and allow the code to be eliminated when the appropriate "ASSUME" defines are set.  (MS Edge is not considered IE by the user agent code).